tensorflow_addons/optimizers/stochastic_weight_averaging.py

```diff
@@ -0,0 +1,242 @@
+            tf.cast(0, tf.int64),
+            tf.math.floordiv(self.iterations - start_averaging,
+                             average_period))
+        # checks if the iteration is one in which a snapshot should be taken.
+        sync_cond = tf.equal(start_averaging + num_snapshots * average_period,
+                             self.iterations)
+        num_snapshots = tf.cast(num_snapshots, tf.float32)
+        average_value = (
+            (average_var * num_snapshots + var) / (num_snapshots + 1.))
+        average_cond = tf.reduce_all([thresold_cond, sync_cond])
+        with tf.control_dependencies([average_value]):
+            average_update = average_var.assign(
+                tf.where(
+                    average_cond,
+                    average_value,
+                    average_var,
+                ),
+                use_locking=self._use_locking)
+        return average_update
+
+    @property
+    def weights(self):
+        return self._weights + self._optimizer.weights
+
+    def _resource_apply_dense(self, grad, var):
+        train_op = self._optimizer._resource_apply_dense(grad, var)  # pylint: disable=protected-access
+        with tf.control_dependencies([train_op]):
+            average_op = self._average_op(var)
+        return tf.group(train_op, average_op)
+
+    def _resource_apply_sparse(self, grad, var, indices):
+        train_op = self._optimizer._resource_apply_sparse(  # pylint: disable=protected-access
+            grad, var, indices)
+        with tf.control_dependencies([train_op]):
+            average_op = self._average_op(var)
+        return tf.group(train_op, average_op)
+
+    def assign_average_vars(self, var_list):
+        """Assign variables in var_list with their respective averages.
+
+        Args:
+            var_list: List of model variables to be assigned to their average.
+
+        Returns:
+            assign_op: The op corresponding to the assignment operation of
+            variables to their average.
+
+        Example:
+        ```python
+        model = tf.Sequential([...])
+        opt = tfa.optimizers.SWA(
+                tf.keras.optimizers.SGD(lr=2.0), 100, 10)
+        model.compile(opt, ...)
+        model.fit(x, y, ...)
+
+        # Update the weights to their mean before saving
+        opt.assign_average_vars(model.variables)
+
+        model.save('model.h5')
+        ```
+        """
+        assign_op = tf.group(
+            [var.assign(self.get_slot(var, 'average')) for var in var_list])
+        return assign_op
+
+    def get_config(self):
+        config = {
+            'optimizer': tf.keras.optimizers.serialize(self._optimizer),
+            'average_period': self._serialize_hyperparameter('average_period'),
+            'start_averaging':
+            self._serialize_hyperparameter('start_averaging')
+        }
+        base_config = super(SWA, self).get_config()
+        return dict(list(base_config.items()) + list(config.items()))
```

Comment:
```suggestion
        return { **base_config, **config }
```

The suggestion is more pythonic(use of dict comprehension) and potentially faster? Since there would be less conversion.